### DISCUSSION TITLE
reason for "end-tag-mismatch" error

### DISCUSSION THREAD

**Discussion by stefanprobst** [mdx-js/mdx#2260] [Q&A] [answered]
hi, i am getting an error with the following simple input, and it's not really clear to me why:

```js
import { compile } from "@mdx-js/mdx";

const input = `<p>
&nbsp;</p>`;

const out = await compile(input);
```

could someone explain? thanks!

interestingly, when i remove `nbsp;` there is no error.

**Comment by ChristianMurphy** [comment, marked as answer]
See https://github.com/mdx-js/mdx/issues/2181 and https://github.com/mdx-js/mdx/issues/2053
To repeat: 

MDX embedded JSX can be inline:
```jsx
<p>text</p>
```

Or MDX embedded JSX can be a block
```jsx
<p>
  text
</p>
```

both of the above are valid.

--- 

Content cannot mix inline and block with the same tag, such as:

```jsx
<p>text
</p>
```

or

```jsx
<p>
text</p>
```

both of the above are invalid
